posthog/models/integration.py

```diff
@@ -1004,15 +1004,40 @@ class GitHubIntegration:
 
     @classmethod
     def client_request(cls, endpoint: str, method: str = "GET") -> requests.Response:
-        jwt_token = jwt.encode(
-            {
-                "iat": int(time.time()) - 300,  # 5 minutes in the past
-                "exp": int(time.time()) + 300,  # 5 minutes in the future
-                "iss": settings.GITHUB_APP_CLIENT_ID,
-            },
-            settings.GITHUB_APP_PRIVATE_KEY,
-            algorithm="RS256",
-        )
+        github_app_client_id = settings.GITHUB_APP_CLIENT_ID
+        github_app_private_key = settings.GITHUB_APP_PRIVATE_KEY
+
+        if not github_app_client_id:
+            raise ValidationError("GITHUB_APP_CLIENT_ID is not configured")
+        
+        if not github_app_private_key:
+            raise ValidationError("GITHUB_APP_PRIVATE_KEY is not configured")
+
+        # Handle common newline encoding issues in environment variables
+        # Replace literal \n with actual newlines
+        github_app_private_key = github_app_private_key.replace('\\n', '\n')
+
+        # Check if the private key has the proper PEM format
+        if not github_app_private_key.strip().startswith("-----BEGIN"):
+            raise ValidationError("GITHUB_APP_PRIVATE_KEY is not in proper PEM format. It should start with -----BEGIN")
+
+        logger.info(f"GITHUB_APP_CLIENT_ID: {github_app_client_id}")
+        logger.info(f"GITHUB_APP_PRIVATE_KEY length: {len(github_app_private_key)}")
+        logger.info(f"GITHUB_APP_PRIVATE_KEY starts with: {github_app_private_key[:50]}...")
```

Comment:
I initially wondered whether we should be logging the private key but I guess it's fine given it's only a portion